library/private/options_tools.h.in

```diff
@@ -100,31 +131,45 @@ ratio_t parse(const std::string& str)
 
 //----------------------------------------------------------------------------
 /**
- * Return provided string
+ * Return provided string stripped of leading and trailing spaces.
  */
 template<>
 std::string parse(const std::string& str)
 {
-  return str;
+  return std::string(options_tools::trim(str));
 }
 
 //----------------------------------------------------------------------------
 /**
- * Parse provided string into a double vector
- * Split the string using ',', then rely on double version for parsing each token
+ * Parse provided string into an int vector using parseVector method
+ * Can throw options::parsing_exception in case of failure to parse
+ */
+template<>
+std::vector<int> parse(const std::string& str)
+{
+  return options_tools::parseVector<int>(str);
+}
```

Comment:
Can you factorize all `std::vector<T> parse(const std::string& str)` functions in the same templated function?

Reply:
I tried this:
```
template<typename T>
std::vector<T> parse(const std::string& str)
{
  return options_tools::parseVector<T>(str);
}
```

But I got:

```
/home/glow/dev/f3d/f3d/build/library/private/options_tools.h:497:82: error: call of overloaded ‘parse<bool>(const std::string&)’ is ambiguous
  497 |   if (name == "interactor.axis") opt.interactor.axis = options_tools::parse<bool>(str);
      |                                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
/home/glow/dev/f3d/f3d/build/library/private/options_tools.h:58:6: note: candidate: ‘T f3d::options_tools::parse(const std::string&) [with T = bool; std::string = std::__cxx11::basic_string<char>’
   58 | bool parse(const std::string& str)
      |      ^~~~~
/home/glow/dev/f3d/f3d/build/library/private/options_tools.h:147:16: note: candidate: ‘std::vector<T> f3d::options_tools::parse(const std::string&) [with T = bool; std::string = std::__cxx11::basic_string<char>]’
  147 | std::vector<T> parse(const std::string& str)

```